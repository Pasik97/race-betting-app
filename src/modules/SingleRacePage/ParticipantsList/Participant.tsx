import React from 'react';

const Participant: React.FC = () => {
   return null;
}

export default Participant;
