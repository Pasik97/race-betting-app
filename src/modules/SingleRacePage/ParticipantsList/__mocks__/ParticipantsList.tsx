const mock = () => '(ParticipantsList /)';

export default mock;
