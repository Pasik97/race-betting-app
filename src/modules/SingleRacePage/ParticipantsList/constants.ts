export interface ParticipantsListProps {
   raceId: string;
};

export interface ParticipantProps {
   participantId: number;
   raceId: string;
}

export const placesEmojis = ['ðŸ¥‡', 'ðŸ¥ˆ', 'ðŸ¥‰'];
