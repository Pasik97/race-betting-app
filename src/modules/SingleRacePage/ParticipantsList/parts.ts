import styled from 'styled-components';
import { Row } from 'components/BasicTable/parts';
import media from 'utils/media';

export const ParticipantsListWrapper = styled.div``;

export const StyledRow = styled(Row)`
   grid-template-columns: 0.5fr 2fr 0.5fr 0.5fr 0.5fr;

   ${media.fromTablet} {
      grid-template-columns: 0.5fr 4fr 0.5fr 0.5fr 0.5fr;
   }

   ${media.fromDesktop} {
      grid-template-columns: 0.5fr 6fr 0.5fr 0.5fr 0.5fr;
   }
`;

export const Radio = styled.input``;
