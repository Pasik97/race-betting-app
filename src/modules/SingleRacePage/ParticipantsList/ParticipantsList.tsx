import React from 'react';
import { useSelector } from 'react-redux';
import * as P from './parts';
import * as TP from 'components/BasicTable/parts';
// import ApplicationState from 'store/ApplicationState';
// import { getRaceById } from 'store/races/selectors';
import { getParticipants } from 'store/participants/selectors';

interface ParticipantsListProps {
   raceId: string;
}

const ParticipantsList: React.FC<ParticipantsListProps> = (({ raceId }) => {
   // const race = useSelector((state: ApplicationState) => getRaceById(state, raceId));
   const participants = useSelector(getParticipants);

   return !participants.length ? null : (
      <P.ParticipantsListWrapper>
         <P.StyledRow>
            <TP.HeaderCell>Id</TP.HeaderCell>
            <TP.HeaderCell>Participant Name</TP.HeaderCell>
            <TP.HeaderCell>ðŸ¥‡</TP.HeaderCell>
            <TP.HeaderCell>ðŸ¥ˆ</TP.HeaderCell>
            <TP.HeaderCell>ðŸ¥‰</TP.HeaderCell>
         </P.StyledRow>
         {/* {filteredRacesOrder?.map((id) => (
            <TP.Row key={races[id].id}>
               <TP.Cell><P.StyledLink to={'/races/' + races[id].id}>{races[id].id}</P.StyledLink></TP.Cell>
               <TP.Cell><P.StyledLink to={'/races/' + races[id].id}>{races[id].name}</P.StyledLink></TP.Cell>
               <TP.Cell><TP.Label>{races[id].active ? 'Active' : 'Inactive'}</TP.Label></TP.Cell>
            </TP.Row>
         ))} */}
      </P.ParticipantsListWrapper>
   );
});

export default ParticipantsList;
