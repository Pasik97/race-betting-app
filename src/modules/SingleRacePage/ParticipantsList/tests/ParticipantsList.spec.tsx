import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import ParticipantsList from "../ParticipantsList";
import { createMockRaceStore } from "modules/SingleRacePage/tests/mockData";
import { mockRace, mockRacesState, secondMockRace } from "store/races/tests/mockData";
import { BrowserRouter } from "react-router-dom";
import { mockParticipantsState } from "store/participants/tests/mockData";

jest.mock('../Participant');

describe('<ParticipantsList />', () => {
   it('should render table header and participant component', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={createMockRaceStore(mockRacesState, mockParticipantsState)}>
               <ParticipantsList raceId={mockRace.id.toString()} />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render null if participants state is empty', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={createMockRaceStore(mockRacesState)}>
               <ParticipantsList raceId={secondMockRace.id.toString()} />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toBeNull();
   });
});
