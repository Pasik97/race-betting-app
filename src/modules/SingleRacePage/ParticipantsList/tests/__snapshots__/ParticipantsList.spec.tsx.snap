// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantsList /> should render table header and participant component 1`] = `
<div
  className="sc-eCImPb"
>
  <div
    className="sc-dkPtRN sc-jRQBWg bmFaAY dgjLOM"
  >
    <div
      className="sc-bdvvtL sc-gsDKAQ YGxmD juxHvi"
    >
      Id
    </div>
    <div
      className="sc-bdvvtL sc-gsDKAQ YGxmD juxHvi"
    >
      Participant Name
    </div>
    <div
      className="sc-bdvvtL sc-gsDKAQ YGxmD juxHvi"
    >
      ğŸ¥‡
    </div>
    <div
      className="sc-bdvvtL sc-gsDKAQ YGxmD juxHvi"
    >
      ğŸ¥ˆ
    </div>
    <div
      className="sc-bdvvtL sc-gsDKAQ YGxmD juxHvi"
    >
      ğŸ¥‰
    </div>
  </div>
  (Participant /)
</div>
`;
